### PR TITLE
BEAM-3361 Increase Go gRPC message size

### PR DESCRIPTION
**Pull request by wcn3**
Increases the buffer for gRPC messages from 4M to 50M.

### CODE REVIEW

**Comment by wcn3** [comment]
r:@herohde

File: sdks/go/pkg/beam/core/runtime/harness/harness.go [hunk 1]
```diff
@@ -243,7 +243,7 @@ func fail(id, format string, args ...interface{}) *fnpb.InstructionResponse {
 func dial(ctx context.Context, endpoint string, timeout time.Duration) (*grpc.ClientConn, error) {
 	log.Infof(ctx, "Connecting via grpc @ %s ...", endpoint)
 
-	opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithBlock()}
+	opts := []grpc.DialOption{grpc.WithInsecure(), grpc.WithBlock(), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(50 << 20))}
```
Comment (herohde) [inline sdks/go/pkg/beam/core/runtime/harness/harness.go]: Please make this change in the grpcx helper as well (which this code is incidentally replaced with in #4265).

**Comment by herohde** [comment]
Minor comment, otherwise LGTM

File: sdks/go/pkg/beam/core/runtime/harness/harness.go (hunk 1 above)
Comment (wcn3) [reply sdks/go/pkg/beam/core/runtime/harness/harness.go]: Done. Thanks for taking care of the timeout change in the new code!

Comment (wcn3) [reply sdks/go/pkg/beam/core/runtime/harness/harness.go]: Something went sideways with my Git. I'm going to abandon this and try again.

**Pull request closed without merging**